app/domain/rotation/master_rotator.rb

```diff
@@ -0,0 +1,93 @@
+module Rotation
+  class MasterRotator
+
+    RotatorNotFound = ::Util::ErrorClass.new(
+      "'{0}' is not an installed rotator"
+    )
+
+    # We inject both the rotation_model and the secret_model to ease unit
+    # testing, and also because, even though the rotation query code was thrown
+    # into Secret, it should probably be refactored out into its own model
+    # class.  When that's done, it will require only a one word change here.
+    #
+    def initialize(rotators:,
+                   account:,
+                   rotation_model: Secret,
+                   secret_model: Secret)
+      @rotators = rotators
+      @account = account
+      @rotation_model = rotation_model
+      @secret_model = secret_model
+    end
+
+    def rotate_every(seconds)
+      loop do
+        rotate_all
+        sleep(seconds)
+      end
+    end
+
+    def rotate_all
+      rotations = rotation_model.required_rotations
+      Parallel.each(rotations, in_threads: 10) do |secret|
+
+        rotator = @rotators[secret['rotator']]
+        raise RotatorNotFound unless rotator
+
+        new_values = rotator.new_values(new_values_input(rotator))
+        update_all_secrets(new_values, secret['ttl'])
+      end
+    end
+
+    private
+
+    def new_values_input(rotator)
```

Comment:
Would this method return a hash of current values that will cumulatively need to be rotated?  Sticking with the AWS example, would it be something like:
```ruby
{
  'cucumber/aws/region': 'us-east-1',
  'cucumber/aws/access_key_id': 'aksdflsdjflkadjflkdjfla1',
  'cucumber/aws/secret_access_key': 'dBQn6phsnbjfZFRC377KShCS29qClGWcEjj6icrojX8=',
}
```

Reply:
Close, but it's only the variable name, not the full resource_id, so it wouldn't include "cucumber."  Since rotators are not account specific, the idea is to clearly separate the concept of "variable names" (what rotators use) and the fully qualified resource_ids (which include account info).  These distinct concepts should probably be reified as objects.